pkg/sshhelper/ssh_auth.go

```diff
@@ -61,6 +70,25 @@ func AddKeyToAgent(keyPath string, a agent.Agent, log logrus.FieldLogger) error
 	if err != nil {
 		return errors.Wrap(err, "failed to read private key")
 	}
+	pubByts, err := ioutil.ReadFile(keyPath + ".pub")
```

Comment:
Can we derive the public key from the private key? I'm weary about forcing a `.pub` to exist, this library is used in a lot of different "environments" at the moment and this could break things

Reply:
Yeah, I wasn't fond of this either, actually looks like that should be doable now that I'm taking a second look at it. I'll fiddle with that.

Reply:
Hmm, I'm not sure if deriving the pubkey is possible without decrypting the private key (providing the passphrase) itself, which is what we're trying to skip. It seems like it should be tho...?

Reply:
From some quick googling https://blog.tinned-software.net/generate-public-ssh-key-from-private-ssh-key/ seems to imply you'd need the passphrase, and this is what the available libraries imply as well.

Reply:
Hmmmm, yes that wouldn't be possible I think. I'd say we'd have to do that here, taking a dependency on `.pub` is to concerning to me at the moment

Reply:
Updated by making this not required in the function